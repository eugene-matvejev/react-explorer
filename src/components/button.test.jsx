import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Button from './button';

configure({ adapter: new Adapter() });

describe('<Button/>', () => {
    const props = {
    };

    describe('render', () => {
        it('with default/require d props', () => {
            const c = shallow(<Button {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<Button {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});

