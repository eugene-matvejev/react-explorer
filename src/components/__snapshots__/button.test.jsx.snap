// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> render with default/require d props 1`] = `
<button
  className="button undefined"
>
   
   
</button>
`;

exports[`<Button/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<button
  className="button undefined"
>
   
   
</button>
`;
